internal/apis/kfd/v1alpha2/eks/create/preflight.go

```diff
@@ -549,3 +560,68 @@ func (p *PreFlight) CheckReducerDiffs(d r3diff.Changelog, diffChecker diffs.Chec
 
 	return nil
 }
+
+func (p *PreFlight) ensureTerraformStateAWSS3Bucket() error {
+	getErr := p.getTerraformStateAWSS3Bucket()
+	if getErr == nil {
+		return nil
+	}
+
+	if errors.Is(getErr, ErrAWSS3BucketNotFound) {
+		if err := p.createTerraformStateAWSS3Bucket(); err != nil {
+			return fmt.Errorf("%w: %w", ErrEnsureTerraformStateAWSS3Bucket, err)
+		}
+
+		return p.getTerraformStateAWSS3Bucket()
+	}
+
+	return getErr
+}
+
+func (p *PreFlight) createTerraformStateAWSS3Bucket() error {
+	bucket := string(p.furyctlConf.Spec.ToolsConfiguration.Terraform.State.S3.BucketName)
+	region := string(p.furyctlConf.Spec.ToolsConfiguration.Terraform.State.S3.Region)
+
+	if _, err := p.awsRunner.S3Api(
+		false,
+		"create-bucket",
+		"--bucket", bucket,
+		"--region", region,
+		"--create-bucket-configuration", fmt.Sprintf("LocationConstraint=%s", region),
+	); err != nil {
+		return fmt.Errorf("%w: %w", ErrCannotCreateTerraformStateAWSS3Bucket, err)
+	}
+
+	return nil
+}
+
+func (p *PreFlight) getTerraformStateAWSS3Bucket() error {
```

Comment:
nit: assertTerraformStateAWSS3BucketMatches or similar instead of get to be consistent with other function names.

Reply:
pushed to develop